go-controller/cmd/ovnkube/ovnkube.go

```diff
@@ -283,12 +283,18 @@ func runOvnKube(ctx *cli.Context) error {
 	// now that ovnkube master/node are running, lets expose the metrics HTTP endpoint if configured
 	// start the prometheus server to serve OVN K8s Metrics (default master port: 9409, node port: 9410)
 	if config.Kubernetes.MetricsBindAddress != "" {
+		if config.Kubernetes.OVNMetricsBindAddress == "" {
+			//Expose OVNRegistry Metrics on same mux as Metrics Server
+
+			//metrics.MakeOvnRegistryPointToDefault()
+			metrics.RegisterOvnMetrics(prometheus.DefaultRegisterer.(*prometheus.Registry), ovnClientset.KubeClient, node)
+		}
 		metrics.StartMetricsServer(config.Kubernetes.MetricsBindAddress, config.Kubernetes.MetricsEnablePprof)
 	}
 
 	// start the prometheus server to serve OVN Metrics (default port: 9476)
 	if config.Kubernetes.OVNMetricsBindAddress != "" {
-		metrics.RegisterOvnMetrics(ovnClientset.KubeClient, node)
+		metrics.RegisterOvnMetrics(metrics.GetOvnRegistry(), ovnClientset.KubeClient, node)
```

Comment:
You can use directly `prometheus.NewRegistry()` here and get rid of `GetOvnRegistry`

Reply:
@fedepaol I was planning to use the same registry as RegisterNorthdMetrics() and RegisterOvnControllerMetrics() for exposing the OvnDB metrics, and hence had the GetOvnRegistry() call. Let me know if this make sense.

Reply:
Yep makes sense